File: app/src/main/java/info/plateaukao/einkbro/unit/RecordUnit.java
package info.plateaukao.einkbro.unit;

import info.plateaukao.einkbro.database.Record;

public class RecordUnit {
    public static final String TABLE_HISTORY = "HISTORY";
    public static final String TABLE_WHITELIST = "WHITELIST";
    public static final String TABLE_JAVASCRIPT = "JAVASCRIPT";
    public static final String TABLE_COOKIE = "COOKIE";

    public static final String COLUMN_TITLE = "TITLE";
    public static final String COLUMN_URL = "URL";
    public static final String COLUMN_TIME = "TIME";
    public static final String COLUMN_DOMAIN = "DOMAIN";

    public static final String CREATE_HISTORY = "CREATE TABLE "
            + TABLE_HISTORY
            + " ("
            + " " + COLUMN_TITLE + " text,"
            + " " + COLUMN_URL + " text,"
            + " " + COLUMN_TIME + " integer"
            + ")";

    public static final String CREATE_WHITELIST = "CREATE TABLE "
            + TABLE_WHITELIST
            + " ("
            + " " + COLUMN_DOMAIN + " text"
            + ")";

    public static final String CREATE_JAVASCRIPT = "CREATE TABLE "
            + TABLE_JAVASCRIPT
            + " ("
            + " " + COLUMN_DOMAIN + " text"
            + ")";

    public static final String CREATE_COOKIE = "CREATE TABLE "
            + TABLE_COOKIE
            + " ("
            + " " + COLUMN_DOMAIN + " text"
            + ")";


    private static Record holder;
    public static Record getHolder() {
        return holder;
    }
    public synchronized static void setHolder(Record record) {
        holder = record;
    }
}

File: app/src/main/java/info/plateaukao/einkbro/unit/RecordUnit.kt
package info.plateaukao.einkbro.unit

object RecordUnit {
    const val TABLE_HISTORY = "HISTORY"
    const val TABLE_WHITELIST = "WHITELIST"
    const val TABLE_JAVASCRIPT = "JAVASCRIPT"
    const val TABLE_COOKIE = "COOKIE"
    const val COLUMN_TITLE = "TITLE"
    const val COLUMN_URL = "URL"
    const val COLUMN_TIME = "TIME"
    const val COLUMN_DOMAIN = "DOMAIN"
    const val CREATE_HISTORY = ("CREATE TABLE "
            + TABLE_HISTORY
            + " ("
            + " " + COLUMN_TITLE + " text,"
            + " " + COLUMN_URL + " text,"
            + " " + COLUMN_TIME + " integer"
            + ")")
    const val CREATE_WHITELIST = ("CREATE TABLE "
            + TABLE_WHITELIST
            + " ("
            + " " + COLUMN_DOMAIN + " text"
            + ")")
    const val CREATE_JAVASCRIPT = ("CREATE TABLE "
            + TABLE_JAVASCRIPT
            + " ("
            + " " + COLUMN_DOMAIN + " text"
            + ")")
    const val CREATE_COOKIE = ("CREATE TABLE "
            + TABLE_COOKIE
            + " ("
            + " " + COLUMN_DOMAIN + " text"
            + ")")
}